src/libDirectoryService/DirectoryService.cpp

```diff
@@ -507,3 +507,82 @@ std::string DirectoryService::GetActionString(Action action) const
         ? "Unknown"
         : ActionStrings.at(action);
 }
+
+uint8_t
+DirectoryService::CalculateNewDifficulty(const uint8_t& currentDifficulty)
+{
+    const uint8_t MAX_ADJUST_STEP = 2;
+    int requiredNodes = 0, powSubmissions = 0;
```

Comment:
consider using uint as requiredNodes and powSubmissions will never be negative

Reply:
Even it will neger go negative, but later need to do a substract, uint substruct uint will never get a minus result.

Reply:
For substract operation, do we need to modify either requiredNodes or powSubmissions?